train/train_process.py

```diff
@@ -173,103 +176,144 @@ def _self_play(self, model_path):
-
-            compare_start_time = time.time()
-            subprocess.run(prepare_cmd(
-                "cargo", "run", "--profile", profile, "-q", "--bin",
-                self.self_play_exec, "--",
-                "--model1-path", best_model,
-                "--model2-path", latest_model,
-                "--games-num", self.cfg["model_compare"]["games_num"],
-                # take the opportunity to generate more games to main games directory
-                "--out-dir1", data_entries_dir,
-                "--out-dir2", tmp_games_dir,
-                "--summary-file", compare_res_file,
-                "--sim-num", self.cfg["mcts"]["sim_num"],
-                "--explore-factor", self.cfg["mcts"]["explore_factor"],
-                "--temperature-policy", self.cfg["model_compare"]["temperature_policy_str"],
-                "--prior-noise-alpha", self.cfg["mcts"]["prior_noise_alpha"],
-                "--prior-noise-epsilon", self.cfg["mcts"]["prior_noise_epsilon"],
-                "--threads", self.cfg["model_compare"]["threads"],
-                "--processing-unit", "CPU" if self.cfg["cpu"] else "GPU"),
-                stderr=sys.stderr, stdout=sys.stdout, shell=True, check=True)
-            self.metrics["model_compare_duration"] = time.time() - compare_start_time
-
-            with open(compare_res_file, "r") as res_file:
-                res = json.load(res_file)
-            w1, w2, d = res["player1_wins"], res["player2_wins"], res["draws"]
-            total_games = w1 + w2 + d
-            winning, losing = w2 / total_games, w1 / total_games
-            self.metrics["trained_model_win_rate"] = winning
-            logging.debug(f"Trained model winning rate: {winning}")
-
-            if winning > self.cfg["model_compare"]["switching_winning_threshold"]:
-                best_model = latest_model
-                # In case the new model is the new best model, take the opportunity and use the new games generated by
-                # the comparison stage as training data in future training steps
-                for filename in os.listdir(tmp_games_dir):
-                    shutil.move(os.path.join(tmp_games_dir, filename), data_entries_dir)
-            elif losing > self.cfg["model_compare"]["warning_losing_threshold"]:
-                logging.warn("New model is worse than previous one, losing ratio: %f", losing)
-
-            return best_model
-        finally:
-            shutil.rmtree(tmp_dirpath)
+            return latest_models[0]
+
+        logging.info("Comparing latest models to best model...")
+        for (model_idx, (latest_model, latest_model_path)) in enumerate(latest_models):
+            tmp_dirpath = tempfile.mkdtemp()
+            try:
+                compare_res_file = os.path.join(tmp_dirpath, "compare_result.json")
+                tmp_games_dir = os.path.join(tmp_dirpath, "games")
+
+                profile = "dev" if self.cfg["debug"] else "release"

```

Comment:
can you move this whole prepare_cmd part into a dedicated function?
just to make this function a bit cleaner

Reply:
done